<?xml version="1.0"?>
<launch>

    <!--Pendulum Parameters-->
    <arg name="friction_coeficient" default="0.01"/>
    <arg name="rod_mass"  default="0.75"/>
    <arg name="rod_length"  default="0.36"/>
    <arg name="gravity"  default="9.8"/>

    <!--Simulation Parameters-->
    <arg name="sample_time" default="0.02"/>
    <arg name="node_rate"  default="150"/>

    <!--Load mesh model -->
    <arg name="urdf_model" value="$(find slm_sim)/urdf/pendulum.urdf"/>
    <param name="robot_description" command="cat $(arg urdf_model)" />

    <!--Robot State Publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="slm_state_pub" >
    </node>

    <!--Robot State Publisher --> <!--Uncomment-->    
    <!--<node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui">
    </node>-->

    <arg name="rvizconfig" default="$(find slm_sim)/rviz/manipulator.rviz" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" /> 

    <node pkg="slm_sim" type="slm_sim.py" name="slm_sim" output="screen">
        <param name="friction_coeficient" value="$(arg friction_coeficient)"/>
        <param name="rod_mass" value="$(arg rod_mass)"/>
        <param name="rod_length" value="$(arg rod_length)"/>
        <param name="gravity" value="$(arg gravity)"/>
        <param name="node_rate" value="$(arg node_rate)"/>
        <param name="sample_time" value="$(arg sample_time)"/>
    </node>

</launch>